File: Codes/1288-remove-covered-intervals_20260708.cpp
/*
 * @Author: LetMeFly
 * @Date: 2026-07-08 11:47:30
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2026-07-08 11:49:29
 */
#ifdef _DEBUG
#include "_[1,2]toVector.h"
#endif

class Solution {
public:
    int removeCoveredIntervals(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end(), [](const vector<int>& a, const vector<int>& b) {
            return a[0] == b[0] ? a[1] > b[1] : a[0] < b[0];
        });
        int ans = intervals.size(), maxr = 0;
        for (vector<int>& v : intervals) {
            ans -= v[1] <= maxr;
            maxr = max(maxr, v[1]);
        }
        return ans;
    }
};

File: Codes/1288-remove-covered-intervals_20260708.py
'''
Author: LetMeFly
Date: 2026-07-08 11:47:30
LastEditors: LetMeFly.xyz
LastEditTime: 2026-07-08 11:50:59
'''
from typing import List

class Solution:
    def removeCoveredIntervals(self, intervals: List[List[int]]) -> int:
        intervals.sort(key=lambda x: (x[0], -x[1]))
        ans = len(intervals)
        maxr = 0
        for _, r in intervals:
            if r <= maxr:
                ans -= 1
            else:
                maxr = r
        return ans
